### PR TITLE
remove hook from helm chart

### PR DESCRIPTION
**Pull request by cwen0**
Signed-off-by: cwen0 <cwenyin0@gmail.com>

### What problem does this PR solve? 

Remove the hook and use `genCA` to generate certs 

### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by cwen0** [comment]
@Gallardot @yeya24 @fewdan PTAL

File: helm/chaos-mesh/templates/webhook-configuration.yaml [hunk 1]
```diff
@@ -1,3 +1,27 @@
+{{- $altNames := list ( printf "%s.%s.svc" (include "chaos-mesh.svc" .) .Release.Namespace )  }}
+{{- $ca := genCA "chaos-mesh-ca" 3650 }}
+{{- $cert := genSignedCert "chaos-mesh-ca" nil $altNames 3650 $ca }}
```
Comment (Gallardot) [inline helm/chaos-mesh/templates/webhook-configuration.yaml:3]: 10 year？
Maybe we should also change the `duration` of `Certificate`

https://github.com/pingcap/chaos-mesh/blob/ad4f1204a410422e0a93f3aba6b143f42cee15df/helm/chaos-mesh/templates/webhook-configuration.yaml#L108

Comment (cwen0) [reply helm/chaos-mesh/templates/webhook-configuration.yaml:3]: got

File: helm/chaos-mesh/README.md [hunk 2]
```diff
@@ -71,7 +71,6 @@ The following tables list the configurable parameters of the Chaos Mesh chart an
 | `preJobs.podAnnotations` | | `{}` |
 | `postJobs.podAnnotations` | | `{}` |
```
Comment (yeya24) [inline helm/chaos-mesh/README.md:72]: Please also delete these two options since they are not used anymore.

**Comment by yeya24** [comment]
Sorry about the flaky E2E, I think we can merge this first because the failed case is not related to this PR. WDYT?

**Pull request merged**